inst/qview/qview.R

```diff
@@ -0,0 +1,194 @@
+#' @importFrom DT dataTableOutput
+#' @export
+queueViewOutput <- function(id) {
+  ns <- NS(id)
+  tagList(
+    DT::dataTableOutput(ns("queue_table"))
+  )
+}
+
+## Server ####
+
+#' queueView - Server-side function to call with callModule()
+#' @param input Shiny input
+#' @param output Shiny output
+#' @param session Shiny session
+#' @param queue_obj An rrsq::RSimpleQueue object that has already been initialized with $new.
+#' @param refresh_interval The frequency with which the UI should poll the queue and update, in milliseconds.
+#' @importFrom rrsq::RSimpleQueue
+#' @export
+queueView <- function(
+    input,
+    output,
+    session,
+    queue_obj = rrsq::RSimpleQueue$new(),
+    .user,
+    .refresh_interval = 3000
+  ) {
+
+  # Grab the namespace function used in the UI object.
+  # We will need this for creating the individual buttons in the list.
+  ns <- session$ns
+
+  print("Starting.")
+  print(glue::glue("User: {.user}"))
+  # browser()
+
+  # Get an RSQ object, which contains the methods needed for interacting with the Queue.
+  queue_obj = rrsq::RSimpleQueue$new()
+
+  # Initialize our data frame, "queue_data".
+  # We pull the source data with queue_obj$get_jobs, then convert it to a data frame and
+  #   sort the contents by ID (descending). We then assign this to as a component of a
+  #   reactiveValue for easy-access in other reactive contexts.
+  .rv <- reactiveValues()
+  .rv$queue_data <-
+    rrsq::jobs_to_df(queue_obj$get_jobs()) %>%
+    dplyr::filter(user == .user) %>%
+    dplyr::arrange(desc(id))
+
+  .return_values <- reactiveValues()
+
+  # Create an observe event that re-runs (via invalidation) every 1000 milliseconds.
+  # In this section, we grab an updated version of our data frame and compare it
+  #   to the one currently in use. If any differences are found, we upate the data
+  #   frame in use. This prevents the UI from constantly refreshing every time it
+  #   checks for new data.
+  observe({
+    invalidateLater(.refresh_interval)
+    new_data <-
+      rrsq::jobs_to_df(queue_obj$get_jobs()) %>%
+      dplyr::filter(user == .user) %>%
+      dplyr::arrange(desc(id))
+
+    if (
+      !isTRUE(all_equal(.rv$queue_data, new_data))
+      ) {
+      .rv$queue_data <- new_data
+    }
+  })
+
+  # Create the DataTable (DT) object to be rendered.
+  #   Using the data in our reactiveValue (.rv$queue_data), we render a Data Table object.
+  #   While doing so, we add a column of buttons to the Data Table with the help of the
+  #   function "cancel_button_creator". "cancel_button_creator" needs our namespacing
+  #   function in order to properly namespace the buttons with this module's UI namespace.
+  output$queue_table  <- DT::renderDT(
+    {
+
+      if(nrow(.rv$queue_data) > 0) {
+        cancel_button_column <- map_chr(
+          1:nrow(.rv$queue_data),
+          cancel_button_creator,
+          .ns = ns
+        )
+      } else {
+        cancel_button_column <- list()
```

Comment:
cancel_button_column is type-unstable in its current form map_chr returns a character _vector_, whereas you'd also have a list if hitting thee else clause. Below in 127, when you pass an empty list to mutate you likely won't get the behavior you want. Instead, you might want, say, an empty row, in which you'd want `cancel_button_column <- ""` The empty string is a vector of length 1, therefore in the mutate call below it will turn each row in that column to an empty string as well.